provider/direct.go

```diff
@@ -37,21 +47,48 @@ func (f CredCheckerFunc) Check(user, password string) (ok bool, err error) {
 	return f(user, password)
 }
 
+// Credentials holds user credentials
+type Credentials struct {
```

Comment:
I don't think this struct supposed to be exposed. If so, making it `credentials` makes more sense